### PR TITLE
Data handler

### PR DESCRIPTION
**Pull request by stefanopolitano**
This PR contains:

first draft of data handler class with relative test function
additional utils function

### CODE REVIEW

File: flarefly/utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self._colour_ = colour
+        
+        if 'OK' in colour:
+            print(f'{Colour_print.OKGREEN}{text}{Colour_print.ENDC}')
+        elif colour == 'WARNING':
+            print(f'{Colour_print.WARNING}{text}{Colour_print.ENDC}')
+        elif colour == 'FAIL':
+            print(f'{Colour_print.FAIL}{text}{Colour_print.ENDC}')
+        else:
+            print(f'{text}')
```
Comment (fgrosa) [inline flarefly/utils.py:34]:
```suggestion
    def __init__(self, text, level):
        """
        Initialize the class
        Parameters
        ------------------------------------------------
        text: str
            Text to be printed
        level: str
            Level of logger, possible values [DEBUG, INFO, WARNING, ERROR, FATAL]
        """
        self._text_ = text
        self._level_ = level
        
        if level == 'DEBUG':
            print(f'{Colour_print.DEBUG}DEBUG{Colour_print.ENDC}: {text}')
        elif level == 'INFO':
            print(f'{Colour_print.INFO}INFO{Colour_print.ENDC}: {text}')
        elif level == 'WARNING':
            print(f'{Colour_print.WARNING}WARNING{Colour_print.ENDC}: {text}')
        elif level == 'ERROR':
            print(f'{Colour_print.ERROR}ERROR{Colour_print.ENDC}: {text}')
        elif level == 'FATAL':
            print(f'{Colour_print.FATAL}FATAL{Colour_print.ENDC}: {text}')
            sys.exit(0)
        else:
            print(text)
```

File: flarefly/utils.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+"""
+
+class Colour_print:
+    """
+    Class to print in colour
+    """    
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
```
Comment (fgrosa) [inline flarefly/utils.py:12]:
```suggestion
    DEBUG = '\033[96m'
    INFO = '\033[92m'
    WARNING = '\033[93m'
    ERROR = '\033[91m'
    FATAL = '\33[101m'
    ENDC = '\033[0m'
```

File: flarefly/utils.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+"""
+Simple module with a class to manage the data used in the analysis
+"""
+
+class Colour_print:
```
Comment (fgrosa) [inline flarefly/utils.py:5]:
```suggestion
#import sys

class Logger:
```

File: test/test_data_handler.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+'''
```
Comment (fgrosa) [inline test/test_data_handler.py:1]: This has to be modified with the fitter class as soon as available (to be kept in mind)

File: flarefly/data_handler.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
+"""
+Simple module with a class to manage the data used in the analysis
+"""
+import sys
+import numpy as np
+import zfit
+import pandas as pd
+import uproot
+from flarefly.utils import Colour_print
```
Comment (fgrosa) [inline flarefly/data_handler.py:9]:
```suggestion
from flarefly.utils import Logger
```

File: flarefly/data_handler.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            if isinstance(data, str):
+                if data.endswith('.root'):
+                    self.__format__ = 'root'
+                    if 'histoname' in kwargs:
+                        histoname = kwargs['histoname']
+                        self._obs_ = zfit.Space(f'{histoname}', limits=(limits[0], limits[1]))
+                    else:
+                        Colour_print('"histoname" not specified. Please specify\
+                                     the name of the histogram to be used', 'FAIL')
+                        sys.exit()
```
Comment (fgrosa) [inline flarefly/data_handler.py:49]:
```suggestion
                        Logger('"histoname" not specified. Please specify the '
                               'name of the histogram to be used', 'FATAL')
```

File: flarefly/data_handler.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                    self._input_ = hist
+                    hist_conv = np.asarray(hist[0], dtype=np.float64)
+                    self._data_ = zfit.data.Data.from_numpy(obs=self._obs_, array=hist_conv)
+                    del hist_conv, hist
+                elif data.endswith('.parquet'):
+                    self.__format__ = 'parquet'
+                    self._data_ = pd.read_parquet(data)
+                else:
+                    Colour_print('Data format not supported yet. Please use .root or .parquet', 'FAIL')
+                    sys.exit()
```
Comment (fgrosa) [inline flarefly/data_handler.py:60]:
```suggestion
                    Logger('Data format not supported yet. Please use .root or .parquet', 'FATAL')
```

File: flarefly/data_handler.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        return self._use_zfit_
+
+    def get_obs(self):
+        """
+        Get the observation space
+        """
+        if self._use_zfit_:
+            return self._obs_
+        else:
+            print('Observable not available for non-zfit data')
```
Comment (fgrosa) [inline flarefly/data_handler.py:113]:
```suggestion
            Logger('Observable not available for non-zfit data', 'ERROR')
```

File: flarefly/data_handler.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        Get the data instance
+        """
+        return self._data_
+
+    def dump_to_pandas(self):
+        """
+        Dump data in pandas df
+        """
+        if self.__format__ == 'pandas':
+            Colour_print('Data already in pandas format.', 'WARNING')
```
Comment (fgrosa) [inline flarefly/data_handler.py:127]:
```suggestion
            Logger('Data already in pandas format.', 'WARNING')
```

File: flarefly/data_handler.py [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        if isinstance(self._input_, np.ndarray):
+            self.__format__ = 'pandas'
+            return self._data_.to_pandas()
+        else:
+            Colour_print('Data format not supported yet for pandas dump.', 'WARNING')
```
Comment (fgrosa) [inline flarefly/data_handler.py:132]:
```suggestion
            Logger('Data format not supported yet for pandas dump.', 'ERROR')
```

**Review by fgrosa — COMMENTED** [review]
Ciao @stefanopolitano, thanks! It looks to me, I just proposed you few small changes in the logger

File: test/test_data_handler.py (hunk 4 above)
Comment (stefanopolitano) [reply test/test_data_handler.py:1]: yes indeed I agree, as soon as we have a first draft of the fitter we can replace this piece

**Review by fgrosa — APPROVED** [review]

**Pull request merged**